R/to_xml.R

```diff
@@ -87,6 +87,22 @@ transform_block <- function(code_block){
   # This prevents partial code blocks that are still apparently valid: ```{r, }
   info <- info[names(info) != ""]
 
+  outchunk_option <- !(names(info) %in% c("language", "name"))
+  names(info)[outchunk_option] <- paste0(names(info)[outchunk_option], "-outchunk")
+
+  code <- strsplit(xml2::xml_text(code_block), "\n")[[1]]
+  inchunk_info <- knitr::partition_chunk(info[["language"]], code)
+  xml2::xml_text(code_block) <- paste0(inchunk_info$code, "\n")
+  inchunk_options <- inchunk_info$options
+
+  inchunk_options <- if (!is.null(inchunk_options) > 0) {
+    yaml::as.yaml(inchunk_options)
```

Comment:
now it's a bit annoying that there's such different treatment for in- and out- chunk options, it feels a bit wrong maybe.

Reply:
Safety first! 🥽 

```suggestion
    yaml::as.yaml(inchunk_options, eval.expr = FALSE)
```

Reply:
This will help parse expressions

```suggestion
    yaml::as.yaml(inchunk_options,
      handlers = list(expression = function(x) paste("!expr: ", paste(as.character(x), sep = "; ")))
```

Reply:
@zkamvar could you please explain this?

Reply:
This is to address expressions that originate from YAML, but I'm now realising that I'm actually having a hard time getting a good example, so I have to rethink this. 

``` r
(x <- list(A = 1, B = "two", C = str2expression("C <- 1 + 2\nC")))
#> $A
#> [1] 1
#> 
#> $B
#> [1] "two"
#> 
#> $C
#> expression(C <- 1 + 2, C)
eval(x$C) # C evaluates to 3
#> [1] 3
try(yaml::as.yaml(x)) # conversion fails because C is an expression
#> Error in yaml::as.yaml(x) : Unknown emitter error
expr_handler <- function(x) paste("!expr:", paste(as.character(x), collapse = "; "))
yaml::as.yaml(x, handlers = list(expression = expr_handler))
#> [1] "A: 1.0\nB: two\nC: '!expr: C <- 1 + 2; C'\n"
```

<sup>Created on 2023-01-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>